muskathlon/models/survey_user_input.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Quentin Gigon <gigon.quentin@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, api
+
+
+class SurveyUserInput(models.Model):
+    _inherit = 'survey.user_input'
+
+    @api.multi
+    def write(self, vals):
+        """
+        Automatically complete Medical Survey task when user filled it
+        """
+        res = super(SurveyUserInput, self).write(vals)
+        if vals.get('state') == 'done':
+
+            # Search for medical surveys
+            for registration in self.env['event.registration'].search([
```

Comment:
You don't need a for loop, as you will only call the write method. You can do instead

```
registrations = self.env[...].search(...)
registrations.write({..})
```